soc/xtensa/intel_adsp/ace/power.c

```diff
@@ -176,6 +176,11 @@ void power_gate_entry(uint32_t core_id)
 	soc_cpus_active[core_id] = false;
 	sys_cache_data_flush_range(soc_cpus_active, sizeof(soc_cpus_active));
 	k_cpu_idle();
+
+	/* It is unlikely we get in here, but when this happens
+	 * we need to lock interruptions again.
+	 */
+	(void)irq_lock();
```

Comment:
Needs to be arch_irq_lock(), or else an RSIL instruction or some other sequence to directly set PS.  When SMP=y, irq_lock() is a global recursive lock that emulates the traditional API for the benefit of drivers not yet ported to spinlocks, not what you want at all (and dangerous if there are other irq_lock() users as it looks to me like this call is unmached and would produce a deadlock if someone else tries it).

Reply:
Ouch forgot about and thought it was just an indirection to `arch_irq_lock`. Yep, it has to be `arch_irq_lock()`, It is rstored later calling directly `rsil`.